File: dataset/output/MISTRAL_codestral-latest/Task75_MISTRAL_codestral-latest.c
// C
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <PID>
", argv[0]);
        return 1;
    }

    int pid = atoi(argv[1]);

    if (pid <= 0) {
        printf("Invalid PID.
");
        return 1;
    }

    if (pid == getpid()) {
        printf("Cannot terminate the current process.
");
        return 1;
    }

    if (kill(pid, SIGTERM) == -1) {
        perror("Error terminating process");
        return 1;
    }

    printf("Process %d terminated.
", pid);
    return 0;
}

File: dataset/output/MISTRAL_codestral-latest/Task75_MISTRAL_codestral-latest.java
// Java
import java.lang.management.ManagementFactory;
import java.lang.management.RuntimeMXBean;
import java.lang.management.StringOperationException;

public class TerminateProcess {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java TerminateProcess <PID>");
            return;
        }

        int pid = Integer.parseInt(args[0]);
        String name = ManagementFactory.getRuntimeMXBean().getName();
        int currentPid = Integer.parseInt(name.split("@")[0]);

        if (pid == currentPid) {
            System.out.println("Cannot terminate the current process.");
            return;
        }

        try {
            Runtime.getRuntime().exec("kill " + pid);
            System.out.println("Process " + pid + " terminated.");
        } catch (Exception e) {
            System.out.println("Error terminating process: " + e.getMessage());
        }
    }
}